app/src/main/kotlin/io/intrepid/skotlinton/base/BasePresenter.kt

```diff
@@ -8,57 +11,31 @@ import io.reactivex.Observable
 import io.reactivex.Scheduler
 import io.reactivex.Single
 import io.reactivex.disposables.CompositeDisposable
+import timber.log.Timber
 
-abstract class BasePresenter<V : BaseContract.View>(protected var view: V?, configuration: PresenterConfiguration) : BaseContract.Presenter {
+abstract class BasePresenter<S : BaseScreen>(protected var screen: S, configuration: PresenterConfiguration) : LifecycleObserver {
 
-    protected val ioScheduler: Scheduler = configuration.ioScheduler
-    protected val uiScheduler: Scheduler = configuration.uiScheduler
+    private val ioScheduler: Scheduler = configuration.ioScheduler
+    private val uiScheduler: Scheduler = configuration.uiScheduler
     protected val userSettings: UserSettings = configuration.userSettings
     protected val restApi: RestApi = configuration.restApi
     protected val crashReporter: CrashReporter = configuration.crashReporter
 
     protected val disposables: CompositeDisposable = CompositeDisposable()
 
-    private var isViewBound = false
-
-    override fun onViewCreated() {
-
-    }
-
-    override fun bindView(view: BaseContract.View) {
-        @Suppress("UNCHECKED_CAST")
-        this.view = view as V
-        if (!isViewBound) {
-            onViewBound()
-            isViewBound = true
-        }
-    }
-
-    open protected fun onViewBound() {
-
+    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
+    fun onCreateScreenLifecycle() {
+        Timber.v("Presenter onCreateScreenLifecycle: $this")
+        screenCreated()
     }
 
-    override fun unbindView() {
+    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+    fun onStopScreenLifecycle() {
         disposables.clear()
-        this.view = null
-
-        if (isViewBound) {
-            onViewUnbound()
-            isViewBound = false
-        }
-    }
-
-    open protected fun onViewUnbound() {
-
+        Timber.v("Presenter onStopScreenLifecycle: $this")
     }
 
-    /**
-     * Note: The view will be null at this point. This method is for any additional cleanup that's not handled
-     * by the CompositeDisposable
-     */
-    override fun onViewDestroyed() {
-
-    }
+    open fun screenCreated() {}
```

Comment:
I think it's better to match Android's usual lifecycle method naming convention and call this `onScreenCreated()`